allennlp/modules/contextual_encoders/contextual_encoder.py

```diff
@@ -0,0 +1,28 @@
+from typing import Dict
+
+import torch
+
+from allennlp.common.registrable import Registrable
+
+class ContextualEncoder(Registrable, torch.nn.Module):
```

Comment:
I know this isn't ready for review yet; feel free to take your time answering this.  I'm just wondering what the difference is between this abstraction and our existing `Seq2SeqEncoder` abstraction.  They look identical to me.

Reply:
you're right, they're mostly identical save a property or two, let me think about how to reuse Seq2SeqEncoder here